Dpdk primary secondary process

ader1990 opened the pull request:
DPDK primary secondary test case required:
  * DPDK 19.05 or greater
  * Linux kernel 4.20 or greater
  * libibverbs version 23 or greater

Currently, libibverbs need to be compiled separately, as there is no support in Azure to install them directly.

v-sijliu commented:
@chvalean, I ran all BVT and WALA testing (which I changed for these cases), 2 failed cases are expected.

Test Results Summary
ICA test run on 11/12/2018 01:49:05
ARM Image under test canonical : ubuntuserver : 18.04-lts : Latest
Total Executed TestCases          18 (         16 Pass,          2 Fail,          0 Abort)
Total Execution Time(dd:hh:mm) 0:1:55
XML file: TestConfiguration

#Sr. Test Name : Test Result : Test Duration (in minutes)
----------------------------------------------------
1. BVT-VERIFY-DEPLOYMENT-PROVISION : PASS : 8.33
	FirstBoot : PASS 
	FirstBoot : Call Trace Verification : PASS 
	Reboot : PASS 
	Reboot : Call Trace Verification : PASS 
2. BVT-IS-ROOT-PASSWORD-DELETED : FAIL : 6.52
3. BVT-ROOT-DEVICE-TIMEOUT-CHECK : PASS : 5.63
4. BVT-HVMODULES-CHECK : PASS : 5.7
5. BVT-SWAP-CHECK : PASS : 5.86
6. BVT-MTAB-ENTRY-CHECK : PASS : 6.36
7. BVT-LAST-CONSOLE-CHECK : PASS : 5.76
8. BVT-VERIFY-BOOT-ERROR-WARNINGS : FAIL : 6.6
9. BVT-VERIFY-SHUTDOWN : PASS : 9.97
10. BVT-VERIFY-RESTART : PASS : 6.26
11. BVT-VERIFY-UUID-FSTAB : PASS : 5.69
12. BVT-VERIFY-SSHD-CONFIG : PASS : 5.88
13. BVT-VERIFY-VHD-PREREQUISITES : PASS : 6.05
14. BVT-VERIFY-NO-ERROR-IN-LOGS : PASS : 5.73
15. BVT-VERIFY-VERBOSE-ENABLED-LOGS : PASS : 6.8
16. BVT-VERIFY-BASH-HISTORY : PASS : 5.7
17. BVT-PASSWD-POLICY-TEST : PASS : 5.65
18. BVT-VERIFY-KVP-DAEMON-STATUS : PASS : 6.5

Test Results Summary
ICA test run on 11/12/2018 01:49:35
ARM Image under test canonical : ubuntuserver : 18.04-lts : Latest
Total Executed TestCases        7 (       7 Pass,        0 Fail,        0 Abort)
Total Execution Time(dd:hh:mm) 0:0:51
XML file: TestConfiguration

#Sr. Test Name : Test Result : Test Duration (in minutes)
----------------------------------------------------
1. BVT-CORE-TIMESYNC-NTP : PASS : 7.09
2. BVT-CORE-VERIFY-LIS-MODULES : PASS : 5.6
3. BVT-CORE-VERIFY-LIS-MODULES-VERSION : PASS : 5.55
4. BVT-CORE-RELOAD-MODULES-SMP : PASS : 16.48
5. BVT-CORE-TIME-CLOCKSOURCE : PASS : 5.43
6. BVT-CORE-TIME-CLOCKEVENT-UP : PASS : 5.53
7. BVT-CORE-TIME-CLOCKEVENT-SMP : PASS : 5.33

Test Results Summary
ICA test run on 11/12/2018 01:50:10
ARM Image under test canonical : ubuntuserver : 18.04-lts : Latest
Total Executed TestCases        7 (       7 Pass,        0 Fail,        0 Abort)
Total Execution Time(dd:hh:mm) 0:0:43
XML file: TestConfiguration

#Sr. Test Name : Test Result : Test Duration (in minutes)
----------------------------------------------------
1. BVT-NET-ETHTOOL-CHECK-KEYS : PASS : 6.06
2. BVT-NET-ETHTOOL-CHANGE-RINGBUFFER : PASS : 5.73
3. BVT-NET-ETHTOOL-TCP-HASHLEVELS : PASS : 5.63
4. BVT-NET-ETHTOOL-UDP-HASHLEVELS : PASS : 5.66
5. BVT-NET-ETHTOOL-GRO-LRO : PASS : 5.65
6. BVT-NET-ETHTOOL-CHECK-CHANNEL : PASS : 5.88
7. BVT-NET-IFUP-IFDOWN : PASS : 8.36

Test Results Summary
ICA test run on 11/12/2018 02:25:46
ARM Image under test canonical : ubuntuserver : 18.04-lts : Latest
Total Executed TestCases         7 (        6 Pass,         0 Fail,         1 Abort)
Total Execution Time(dd:hh:mm) 0:0:44
XML file: TestConfiguration

#Sr. Test Name : Test Result : Test Duration (in minutes)
----------------------------------------------------
1. WALA-VERIFY-HOSTNAME : PASS : 8.33
2. WALA-PROCESS-CHECK : PASS : 5.83
3. WALA-VERSION-CHECK : PASS : 7.5
4. WALA-RESOURCE-DISK-FILESYSTEM-CHECK : PASS : 5.69
5. WALA-VERIFY-MNT-RESOURCE-WRITABLE : PASS : 5.76
6. WALA-VERIFY-MNT-RESOURCE-README : PASS : 5.78
7. WALA-VERIFY-FIREWALL-STATUS : PASS : 5.52

ader1990 committed:
dpdk: added primary secondary processes test case

DPDK primary secondary test case required:

DPDK 19.05 or greater
Linux kernel 4.20 or greater
libibverbs version 23 or greater
Currently, libibverbs need to be compiled separately, as there is no
support in Azure to install them directly.

The test case somtimes fails because the client (secondary) process
can crash because of a segmentation fault. This failure will be
fixed on the DPDK master tree.
